File: src/app/Models/Task.ts
/// Sqlite cannot store dates so we use a string for dueDate and convert it to a date when we need it
export type Task = {
    id: number;
    name: string;
    dueDate: string;
    completed: boolean;
    text: string;
}

export type InsertTask = {
    name: string;
    dueDate: string;
    completed: boolean;
    text: string;
}

File: src-tauri/src/Models/task.rs
use chrono::naive::NaiveDate;

pub struct Task {
    id: i32,
    name: string,
    due_date: NaiveDate,
    completed: bool,
    text: string,
}

pub struct TaskInsert {
    name: string,
    due_date: NaiveDate,
    completed: bool,
    text: string,
}
